src/lib/core/a11y/interactivity-checker.spec.ts

```diff
@@ -234,63 +237,274 @@ describe('InteractivityChecker', () => {
+
+      it('should never mark object frame elements as tabbable', () => {
+        let objectEl = createFromTemplate('<object>', true);
+
+        expect(checker.isTabbable(objectEl)).toBe(false);
+      });
+
+    }));
+
+    describe('for Blink browsers', runIf(platform.BLINK, () => {
+
+      it('should always mark audio elements with controls as tabbable', () => {
+        let audio = createFromTemplate('<audio controls>', true);
+
+        expect(checker.isTabbable(audio)).toBe(true);
+
+        audio.tabIndex = -1;
+
+        // The audio element will be still tabbable because Blink always
+        // considers them as tabbable.
+        expect(checker.isTabbable(audio)).toBe(true);
+      });
+
+    }));
+
+    describe('for Internet Explorer', runIf(platform.TRIDENT, () => {
+
+      it('should never mark video elements without controls as tabbable', () => {
+        // In Internet Explorer video elements without controls are never tabbable.
+        let video = createFromTemplate('<video>', true);
+
+        expect(checker.isTabbable(video)).toBe(false);
+
+        video.tabIndex = 0;
+
+        expect(checker.isTabbable(video)).toBe(false);
+
+      });
+
+    }));
+
+    describe('for iOS browsers', runIf(platform.IOS && platform.WEBKIT, () => {
+
+      it('should never allow div elements to be tabbable', () => {
+        let divEl = createFromTemplate('<div tabindex="0">', true);
+
+        expect(checker.isTabbable(divEl)).toBe(false);
+      });
+
+      it('should never allow span elements to be tabbable', () => {
+        let spanEl = createFromTemplate('<span tabindex="0">Text</span>', true);
+
+        expect(checker.isTabbable(spanEl)).toBe(false);
+      });
+
+      it('should never allow button elements to be tabbable', () => {
+        let buttonEl = createFromTemplate('<button tabindex="0">', true);
+
+        expect(checker.isTabbable(buttonEl)).toBe(false);
+      });
+
+      it('should never allow anchor elements to be tabbable', () => {
+        let anchorEl = createFromTemplate('<a tabindex="0">Link</a>', true);
+
+        expect(checker.isTabbable(anchorEl)).toBe(false);
+      });
+
+    }));
+
+
   });
 
   /** Creates an array of elements with the given node names. */
   function createElements(...nodeNames: string[]) {
     return nodeNames.map(name => document.createElement(name));
   }
 
+  function createFromTemplate(template: string, append = false) {
+    let tmpRoot = document.createElement('div');
+    tmpRoot.innerHTML = template;
+
+    let element = tmpRoot.firstElementChild;
+
+    tmpRoot.removeChild(element);
+
+    if (append) {
+      appendElements([element]);
+    }
+
+    return element as HTMLElement;
+  }
+
   /** Appends elements to the testContainerElement. */
   function appendElements(elements: Element[]) {
     for (let e of elements) {
       testContainerElement.appendChild(e);
     }
   }
+
+  function runIf(condition: boolean, runFn: Function): () => void {
+    return function() {
```

Comment:
We need to use a normal function because we want to retrieve the `arguments` below.